crates/wasi-provider/src/states/starting.rs

```diff
@@ -109,14 +108,14 @@ impl State<PodState> for Starting {
 
         info!("Starting containers for pod {:?}", pod.name());
         for container in pod.containers() {
-            let container_handle = match start_container(pod_state, &pod, &container).await {
-                Ok(h) => h,
-                Err(e) => fail_fatal!(e),
-            };
-            container_handles.insert(
-                ContainerKey::App(container.name().to_string()),
-                container_handle,
-            );
+            match start_container(pod_state, &pod, &container).await {
+                Ok(h) => {
+                    container_handles.insert(ContainerKey::App(container.name().to_string()), h);
+                }
+                // We should log, transition to running, and properly handle container failure.
```

Comment:
This is somewhat tied to making all of these handlers infallible, but this one in particular looked like it would drop important container status patches. The same can be said of the `Initializing` state handler.